py3-bokeh multiversion

dannf opened the pull request:
Add multiversion support to `py3-bokeh`

smoser committed:
Drop simple 'import' tests that are to be replaced.

dannf committed:
py3-bokeh multiversion

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

smoser commented:
Just repeating here so you see it.  I have a branch of files converted to mult-version committed that you can start from.
it may or may not help you learn.
https://github.com/chainguard-dev/internal-dev/issues/5334#issuecomment-2460116967

dannf committed:
Add py3-setuptools-git-versioning

This is a dependency for py3-bokeh. py3-bokeh has been successfully
building without it but, once we migrate it to the
py/pip-build-install pipeline, bokeh will get installed with a
bogus version (0.0.0) if this dep is not present to dynamically
populate the version.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

smoser commented on py3-bokeh.yaml:
```diff
@@ -38,8 +107,6 @@ pipeline:
     uses: python/build-wheel
 
-  - uses: strip
-
```
i think we're leaving these in. they don't do anything if there aren'[t c modules, but its not harmful (and if it was, we need to fix it).

smoser reviewed (changes requested):
one change request.

dannf commented on py3-setuptools-git-versioning.yaml:
```diff
@@ -0,0 +1,113 @@
+      tag: v${{package.version}}
+
+  - name: Python Build
+    uses: python/build-wheel
```
nice catch!

smoser commented on py3-setuptools-git-versioning.yaml:
```diff
@@ -0,0 +1,114 @@
+      tag: v${{package.version}}
+
+  - name: Python Build
+    uses: py/pip-build-install
```
```yaml
        with:
          python: python${{range.key}}
```

dannf replied:
Actually, do we just need to kill this section now that the builds are happening in the `subpackages:` ?

smoser replied:
actually, yes. good catch.

smoser replied:
and can you move the pipeline above the subpackages ? i really wich 'yam' would do all the things, but it does not.

dannf replied:
Could we add a jsonschema for the YAML, and and a `pre-commit` hook to validate it? That could enforce ordering, as well as using `yam`

smoser replied:
huh. i did not realize jsonschema could enforce ordering. I opened https://github.com/chainguard-dev/yam/issues/71 a while ago, I think people would generally be open to computer-enforced formatting (ie, 'go fmt').

dannf replied:
You're right - `jsonschema` doesn't enforce ordering (at least, not AFAIK). What I should've said is that I've written `pre-commit` hooks to enforce that YAML keys are sorted in the order that they are defined in a `jsonschema` file. I suppose the schema isn't strictly needed for the sorting, but schema validation is useful as well.

smoser reviewed (approved):
Thanks

smoser reviewed (changes requested):
actually. i'm going to ask one more change.
Can you add a variable 'import' and use it?

All of the recently modified py3-* files are using that.  (see [py3-absl-py](https://github.com/wolfi-dev/os/blob/main/py3-absl-py.yaml) for example).  That just gives us better data to work with if we're refactoring things later.

dannf committed:
py3-bokeh: Define and use ${{vars.import}}

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>
